internal/anchore/client.go

```diff
@@ -56,3 +57,27 @@ func (c *Client) newRequestContext(parentContext context.Context) context.Contex
 		},
 	)
 }
+
+func hasScheme(url string) bool {
+	parts := strings.Split(url, "://")
+
+	return len(parts) > 1
+}
+
+func ensureURLHasScheme(url string) string {
+	const defaultScheme = "http"
+
+	if !hasScheme(url) {
+		return fmt.Sprintf("%s://%s", defaultScheme, url)
+	}
+
+	return url
+}
+
+func ensureURLHasSuffix(url, suffix string) string {
+	if !strings.HasSuffix(url, suffix) {
+		return url + suffix
```

Comment:
this doesn't look robust to support trailing slashes with leading slashes. Seems like Go can handle this better with `url.Parse` and `path.Join`

```go
package main

import (
	"fmt"
	"log"
	"net/url"
	"path"
)

func main() {
	u, err := url.Parse("http://localhost/")
	if err != nil {
		log.Fatal(err)
	}
	u.Path = path.Join(u.Path, "/v1/")
	s := u.String()
	fmt.Println(s)
}
```

Prints:

```
http://localhost/v1
```

Reply:
See [other comment](https://github.com/anchore/syft/pull/318#discussion_r568189105). We _are_ sanitizing slashes, just not in this discrete operation.

Reply:
I think that preventing these slashes from creating a problem is something that we should try - even though this is not the exact problem this PR is addressing. Path normalization when adding parts is common enough (Python has the same facilities), and I think we can take advantage of it in this case.

Reply:
>I think that preventing these slashes from creating a problem is something that we should try

I completely agree. I think what I'm trying to communicate is that it seems like we're doing this more than you're realizing. It seems like you're focusing specifically in the function `ensureURLHasSuffix`, but there's more to the URL parsing than this function. If you look at how this function is used in context, I think you'll see a bigger picture.

Another way to frame this is to ask the question: "What's some input that produces a bad or unexpected result?"

If we can't think of any cases, then we'd have to conclude that the logic actually is robust.

Having said all that, I think we could rearrange the functionality. Let's pair on this tomorrow!